osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -317,5 +350,42 @@ public BindableStarDifficulty(IBeatmapInfo beatmapInfo, CancellationToken cancel
                 CancellationToken = cancellationToken;
             }
         }
+
+        /// <summary>
+        /// A working beatmap that caches its playable representation.
+        /// This is intended as single-use for when it is guaranteed that the playable beatmap can be reused.
+        /// </summary>
+        private class PlayableCachedWorkingBeatmap : IWorkingBeatmap
```

Comment:
This is a hack because `DifficultyCalculator` only accepts an `IWorkingBeatmap` and I want to avoid double-fetching the playable beatmap. I'm not sure if there's a future where `DifficultyCalculator` accepts a playable beatmap, but it can be experimented with when it gets rewritten (something I'm in the progress of).